File: backend/app/schemas/incident.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from uuid import UUID
from app.models.incident import IncidentType, Severity, IncidentStatus

class IncidentBase(BaseModel):
    type: IncidentType
    severity: Severity
    latitude: float
    longitude: float
    address: Optional[str] = None
    description: Optional[str] = None
    confidence_score: Optional[float] = None
    video_url: Optional[str] = None
    thumbnail_url: Optional[str] = None

class IncidentCreate(IncidentBase):
    pass

class Incident(IncidentBase):
    id: UUID
    status: IncidentStatus
    created_at: datetime
    updated_at: Optional[datetime] = None
    resolved_at: Optional[datetime] = None
    reported_by: Optional[UUID] = None
    verified_by: Optional[UUID] = None


    class Config:
        from_attributes = True

File: mobile_app/lib/models/incident.dart
class Incident {
  final int id;
  final String type;
  final double latitude;
  final double longitude;
  final String? description;
  final String? videoUrl;
  final DateTime timestamp;
  final String status;

  Incident({
    required this.id,
    required this.type,
    required this.latitude,
    required this.longitude,
    this.description,
    this.videoUrl,
    required this.timestamp,
    required this.status,
  });

  factory Incident.fromJson(Map<String, dynamic> json) {
    return Incident(
      id: json['id'],
      type: json['type'],
      latitude: json['latitude'],
      longitude: json['longitude'],
      description: json['description'],
      videoUrl: json['video_url'],
      timestamp: DateTime.parse(json['timestamp']),
      status: json['status'],
    );
  }
}
